nixos/modules/services/hardware/nvidia-container-toolkit/default.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+
+{
+
+  options = {
+
+    hardware.nvidia-container-toolkit.enable = lib.mkOption {
+      default = false;
+      type = lib.types.bool;
+      description = lib.mdDoc ''
+        Enable dynamic CDI configuration for NVidia devices by running
+        nvidia-container-toolkit on boot.
+      '';
+    };
+
+  };
+
+  config = {
+
+    systemd.services.nvidia-container-toolkit = lib.mkIf config.hardware.nvidia-container-toolkit.enable {
+      description = "Container Device Interface (CDI) for Nvidia generator";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "systemd-udev-settle.service" ];
+      serviceConfig = {
+        RuntimeDirectory = "cdi";
+        RemainAfterExit = true;
+        ExecStart = let
+          script = (pkgs.writeScriptBin "nvidia-cdi-generator"
+            (import ./cdi-generate.nix { inherit config pkgs; })); in (lib.getExe script);
+        ExecStop = let
+          script = pkgs.writeScriptBin "nvidia-cdi-generator-cleanup" ''
+            #! ${pkgs.runtimeShell}
+            rm $RUNTIME_DIRECTORY/nvidia-container-toolkit.json
+          ''; in (lib.getExe script);
```

Comment:
Oh... is this necessary because of `RemainAfterExit`?

Reply:
Yes, exactly.

Reply:
Actually, I think it still shouldn't be. Here's my guess:

1. The unit starts (e.g. based on `wantedBy`) and directory is created prior to `ExecStart`
2. The process exits, but the state transitions to `active` because of `RemainAfterExit`
3. Once you `systemctl stop ...` the state transitions to deactivating/dead and systemd should still remove the runtime directory

Reply:
> Yes, exactly.

Interesting

Reply:
I see that it's behaving as you describe, removing it. Not sure what was the condition I was looking at when I had to add it. Anyway...